### PR TITLE
Refactor project snapshot manager

### PR DESCRIPTION
**Pull request by rynowak**
Splits the 'trigger' out from the change manager. The next change will
add more functionality to DPSMBase.

### CODE REVIEW

File: test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Xunit/ForegroundTheoryAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Test.Xunit
```
Comment (ajaybhargavb) [inline test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Xunit/ForegroundTheoryAttribute.cs]: Not sure why these changes are displayed here while the base branch is set to `rynowak/foreground-theory`

Comment (rynowak) [reply test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Xunit/ForegroundTheoryAttribute.cs]: github doesn't update the diff when you push to the base branch

**Comment by rynowak** [comment]
82866d9442ae7914980576fd821e162d6c30ecbb
